src/Agent.Worker/Build/BuildDirectoryManager.cs

```diff
@@ -81,19 +81,25 @@ public sealed class BuildDirectoryManager : AgentService, IBuildDirectoryManager
 
         public TrackingConfig PrepareDirectory(
             IExecutionContext executionContext,
-            RepositoryResource repository,
+            IList<RepositoryResource> repositories,
             WorkspaceOptions workspace)
         {
             // Validate parameters.
             Trace.Entering();
             ArgUtil.NotNull(executionContext, nameof(executionContext));
             ArgUtil.NotNull(executionContext.Variables, nameof(executionContext.Variables));
-            ArgUtil.NotNull(repository, nameof(repository));
+            ArgUtil.NotNull(repositories, nameof(repositories));
+
+            // Get the primary repository (self)
+            var primaryRepository = RepositoryUtil.GetRepository(repositories);
```

Comment:
not a big deal, and definitely not something to hold up this PR for, but would this read better if RepositoryUtil.GetRepository() required repoAlias to be passed in and there was another method RepositoryUtil.GetPrimaryRepository() that just called GetRepository with PrimaryRepositoryName?

Reply:
That's good feedback. I will consider that in my next PR.